CI crashes in hibernate/cache-v62 test

### Describe the bug

The CI crashes when running the tests in the module `hibernate/cache-v62`. The CI message includes:

```
2025-04-16T01:21:15.6886217Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.5.3:test (hibernate-6.6) on project infinispan-hibernate-cache-v62: 
2025-04-16T01:21:15.6887455Z [ERROR] 
2025-04-16T01:21:15.6888619Z [ERROR] See /home/runner/work/infinispan/infinispan/test_dir/infinispan/hibernate/cache-v62/target/surefire-reports for the individual test results.
2025-04-16T01:21:15.6890217Z [ERROR] See dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
2025-04-16T01:21:15.6891149Z [ERROR] There was an error in the forked process
2025-04-16T01:21:15.6892409Z [ERROR] The wrapped NoSuchMethodError is likely caused by the versions of JUnit jars on the classpath/module path not being properly aligned. 
2025-04-16T01:21:15.6894104Z [ERROR] Please ensure consistent versions are used (see https://junit.org/junit5/docs/5.12.1/user-guide/#dependency-metadata).
2025-04-16T01:21:15.6895595Z [ERROR] The following conflicting versions were detected:
2025-04-16T01:21:15.6896329Z [ERROR] - org.junit.jupiter.api: 5.12.2
2025-04-16T01:21:15.6896975Z [ERROR] - org.junit.jupiter.engine: 5.12.2
2025-04-16T01:21:15.6897801Z [ERROR] - org.junit.platform.commons: 1.12.2
2025-04-16T01:21:15.6898474Z [ERROR] - org.junit.platform.engine: 1.12.2
2025-04-16T01:21:15.6899145Z [ERROR] - org.junit.platform.launcher: 1.12.1
2025-04-16T01:21:15.6899826Z [ERROR] - org.junit.vintage.engine: 5.12.2
2025-04-16T01:21:15.6900416Z [ERROR] 
2025-04-16T01:21:15.6901733Z [ERROR] org.junit.platform.commons.JUnitException: The wrapped NoSuchMethodError is likely caused by the versions of JUnit jars on the classpath/module path not being properly aligned. 
2025-04-16T01:21:15.6903687Z [ERROR] Please ensure consistent versions are used (see https://junit.org/junit5/docs/5.12.1/user-guide/#dependency-metadata).
2025-04-16T01:21:15.6905111Z [ERROR] The following conflicting versions were detected:
2025-04-16T01:21:15.6905866Z [ERROR] - org.junit.jupiter.api: 5.12.2
2025-04-16T01:21:15.6906533Z [ERROR] - org.junit.jupiter.engine: 5.12.2
2025-04-16T01:21:15.6907237Z [ERROR] - org.junit.platform.commons: 1.12.2
2025-04-16T01:21:15.6907941Z [ERROR] - org.junit.platform.engine: 1.12.2
2025-04-16T01:21:15.6908620Z [ERROR] - org.junit.platform.launcher: 1.12.1
2025-04-16T01:21:15.6909318Z [ERROR] - org.junit.vintage.engine: 5.12.2
2025-04-16T01:21:15.6910112Z [ERROR] 
2025-04-16T01:21:15.6911088Z [ERROR] 	at org.junit.platform.launcher.core.ClasspathAlignmentChecker.check(ClasspathAlignmentChecker.java:96)
2025-04-16T01:21:15.6912705Z [ERROR] 	at org.junit.platform.launcher.core.ClasspathAlignmentChecker.check(ClasspathAlignmentChecker.java:60)
2025-04-16T01:21:15.6914514Z [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:161)
2025-04-16T01:21:15.6916641Z [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:132)
2025-04-16T01:21:15.6918416Z [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:105)
2025-04-16T01:21:15.6920149Z [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:78)
2025-04-16T01:21:15.6921693Z [ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:99)
2025-04-16T01:21:15.6923013Z [ERROR] 	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:77)
2025-04-16T01:21:15.6924416Z [ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
2025-04-16T01:21:15.6926377Z [ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.lambda$discover$0(InterceptingLauncher.java:33)
2025-04-16T01:21:15.6928589Z [ERROR] 	at org.junit.platform.launcher.core.ClasspathAlignmentCheckingLauncherInterceptor.intercept(ClasspathAlignmentCheckingLauncherInterceptor.java:25)
2025-04-16T01:21:15.6930512Z [ERROR] 	at org.junit.platform.launcher.core.InterceptingLauncher.discover(InterceptingLauncher.java:33)
2025-04-16T01:21:15.6931932Z [ERROR] 	at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
2025-04-16T01:21:15.6933308Z [ERROR] 	at org.apache.maven.surefire.junitplatform.LazyLauncher.discover(LazyLauncher.java:50)
2025-04-16T01:21:15.6934963Z [ERROR] 	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:52)
2025-04-16T01:21:15.6936475Z [ERROR] 	at org.apache.maven.surefire.api.util.DefaultScanResult.applyFilter(DefaultScanResult.java:87)
2025-04-16T01:21:15.6938119Z [ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:144)
2025-04-16T01:21:15.6939835Z [ERROR] 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:124)
2025-04-16T01:21:15.6941350Z [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
2025-04-16T01:21:15.6942681Z [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
2025-04-16T01:21:15.6944001Z [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
2025-04-16T01:21:15.6945331Z [ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
2025-04-16T01:21:15.6947120Z [ERROR] Caused by: java.lang.NoSuchMethodError: 'void org.junit.jupiter.engine.config.DefaultJupiterConfiguration.<init>(org.junit.platform.engine.ConfigurationParameters)'
2025-04-16T01:21:15.6949374Z [ERROR] 	at org.hibernate.testing.bytecode.enhancement.extension.engine.BytecodeEnhancedTestEngine.discover(BytecodeEnhancedTestEngine.java:66)
2025-04-16T01:21:15.6951430Z [ERROR] 	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:152)
2025-04-16T01:21:15.6952658Z [ERROR] 	... 19 more
```

### Expected behavior

CI should be completed successfully.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/hibernate/cache-v62/pom.xml
+++ b/hibernate/cache-v62/pom.xml
@@ -101,9 +101,10 @@
             </configuration>
             <dependencies>
                <dependency>
-                  <groupId>org.junit.vintage</groupId>
-                  <artifactId>junit-vintage-engine</artifactId>
+                  <groupId>org.junit</groupId>
+                  <artifactId>junit-bom</artifactId>
                   <version>${version.junit5}</version>
+                  <type>pom</type>
                </dependency>
             </dependencies>
             <executions>
@@ -181,6 +182,10 @@
                               <classpathDependencyExclude>io.smallrye:jandex</classpathDependencyExclude>
                               <classpathDependencyExclude>org.glassfish.jaxb:jaxb-core</classpathDependencyExclude>
                               <classpathDependencyExclude>org.glassfish.jaxb:txw2</classpathDependencyExclude>
+                              <classpathDependencyExclude>jakarta.activation:jakarta.activation-api</classpathDependencyExclude>
+                              <classpathDependencyExclude>net.bytebuddy:byte-buddy</classpathDependencyExclude>
+                              <classpathDependencyExclude>org.eclipse.angus:angus-activation</classpathDependencyExclude>
+                              <classpathDependencyExclude>org.jboss.logging:jboss-logging</classpathDependencyExclude>
                            </classpathDependencyExcludes>
                            <additionalClasspathDependencies>
                               <additionalClasspathDependency>
@@ -220,13 +225,13 @@
                               <classpathDependencyExclude>io.smallrye:jandex</classpathDependencyExclude>
                               <classpathDependencyExclude>org.glassfish.jaxb:jaxb-core</classpathDependencyExclude>
                               <classpathDependencyExclude>org.glassfish.jaxb:txw2</classpathDependencyExclude>
+                              <classpathDependencyExclude>jakarta.activation:jakarta.activation-api</classpathDependencyExclude>
+                              <classpathDependencyExclude>net.bytebuddy:byte-buddy</classpathDependencyExclude>
+                              <classpathDependencyExclude>org.eclipse.angus:angus-activation</classpathDependencyExclude>
+                              <classpathDependencyExclude>org.jboss.logging:jboss-logging</classpathDependencyExclude>
+                              <classpathDependencyExclude>org.hibernate.common:hibernate-commons-annotations</classpathDependencyExclude>
                            </classpathDependencyExcludes>
                            <additionalClasspathDependencies>
-                              <additionalClasspathDependency>
-                                 <groupId>org.junit.jupiter</groupId>
-                                 <artifactId>junit-jupiter-engine</artifactId>
-                                 <version>${version.junit5}</version>
-                              </additionalClasspathDependency>
                               <additionalClasspathDependency>
                                  <groupId>org.hibernate.orm</groupId>
                                  <artifactId>hibernate-core</artifactId>

